### PR TITLE
Add sigprocmask

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**

### CODE REVIEW

File: src/sys/signal.rs [hunk 1]
```diff
@@ -445,6 +445,27 @@ pub fn pthread_sigmask(how: SigmaskHow,
     Errno::result(res).map(drop)
 }
 
+/// Examine and change blocked signals.
+///
+/// For more informations see the [`sigprocmask` man
+/// pages](http://man7.org/linux/man-pages/man2/sigprocmask.2.html).
```
Comment (asomers) [inline src/sys/signal.rs]: This is a portable function, so you should link to a portable man page.
http://pubs.opengroup.org/onlinepubs/9699919799/functions/sigprocmask.html

File: test/sys/test_signal.rs [hunk 2]
```diff
@@ -5,3 +5,41 @@ use nix::sys::signal::*;
...
 }
+
+#[test]
+fn test_sigprocmask_noop() {
+    sigprocmask(SigmaskHow::SIG_BLOCK, None, None)
+        .expect("this should be an effective noop");
+}
+
+#[test]
+fn test_sigprocmask() {
```
Comment (asomers) [inline test/sys/test_signal.rs]: You should grab `SIGNAL_MTX` here so that this doesn't race with any other tests.

File: test/sys/test_signal.rs [hunk 3]
```diff
@@ -5,3 +5,41 @@ use nix::sys::signal::*;
...
+                different one", SIGNAL);
+
+    // Now block the signal.
+    let mut signal_set = SigSet::empty();
+    signal_set.add(SIGNAL);
+    sigprocmask(SigmaskHow::SIG_BLOCK, Some(&signal_set), None)
+        .expect("expect to be able to block signals");
+
+    // And test it again, to make sure the change was effective.
+    sigprocmask(SigmaskHow::SIG_BLOCK, None, Some(&mut old_signal_set))
```
Comment (asomers) [inline test/sys/test_signal.rs]: How about emptying `old_signal_set` before this line?

**Comment by Thomasdezeeuw** [comment]
Addressed all feedback, you might want to squash the commits since the second doesn't have a good commit message.

**Comment by Susurrus** [comment]
@Thomasdezeeuw Please go ahead and squash the commits on your end and force-push back to this branch.

**Comment by Thomasdezeeuw** [comment]
@Susurrus can't you use the squash and merge button on GitHub? Force pushing to Github never seems to work for me.

**Comment by Susurrus** [comment]
We don't use GitHub to merge PRs, we rely on Bors, an external tool. It doesn't squash commits. So we rely on the PR authors to have their commits in order before we merge it.

**Comment by Thomasdezeeuw** [comment]
@Susurrus I didn't know that, squashed the commits so this is ready to go.

**Comment by Susurrus** [comment]
It's somewhat mentioned in the CONYRIBUTING.MD document, but not very well. I've been meaning to update that actually as this comes up a lot.

**Comment by Thomasdezeeuw** [comment]
It seems the build bot is failing, it times out after 5 minutes and then fails to kill it.

```
command timed out: 300 seconds without output running ['rustup', 'run', '1.20.0', 'cargo', 'test'], attempting to kill
SIGKILL failed to kill process
using fake rc=-1
program finished with exit code -1
remoteFailed: [Failure instance: Traceback from remote host -- exceptions.RuntimeError: SIGKILL failed to kill process
]


buildbot.process.remotecommand.RemoteException: exceptions.RuntimeError: SIGKILL failed to kill process
Traceback (most recent call last):
Failure: exceptions.RuntimeError: SIGKILL failed to kill process
```

And trying to digg in further fails to load (part of) the page.

**Comment by asomers** [comment]
It looks like my VM suddenly got about 10 times slower.  Problems with the VM host?  Meltdown/Spectra countermeasures?  I don't know.  I've bumped the timeouts and I'm restarting the builds.

**Pull request merged**